### DISCUSSION TITLE
Is there an option to combine mask and placeholder? (example inside)

### DISCUSSION THREAD

**Discussion by nikriaz** [beholdr/maska#127] [Q&A]
Example:
For the phone input mask +1 (###)..., user starts to type **2** and gets **+1 (2**

Would be possible to show to user the entire placeholder, not just already completed part? E.g.:
user starts to type **2** and gets **+1 (2  )** and user continues to type area code _inside_ the brackets. (BTW, as you can see, the space occupied by regular spaces does not equal to the space of characters, so we probably need special Unicode spaces here).

Honestly, the "standard" way is very easy to implement without any plugins, it's just few lines of code. The latter example is way more complex and I can't find a ready-to-use library for that. If Maska can't do that, could you please share your thoughts why and possible better way to implement such feature?

**Comment by beholdr** [comment]
Not sure what you mean by «standard» way, but I doubt the need to implement this functionality because I want to keep Maska small and simple.

**Reply by nikriaz** [reply]
"Standard" way is what Maska, as well as many other plugins, do it now. There is [one](https://github.com/RobinHerbots/Inputmask), though, which shows replaceable placeholders on the input. 

It's not ideal too, because it forces user to consequently type correct input from the keyboard. In real life people often paste incomplete or type incorrect input right to the field and then make corrections there. So ideal mask would be to allow user to input anything but detects correct pattern and as soon as it's detected, shows placeholders. Just thoughts.

**Reply by beholdr** [reply]
There are many mask libraries out there. Some are fully featured, others are minimal. Library that you linked weighs ten times as much as Maska. Obviously it has more features, but not everyone needs them. And if you need them, why don't just use that library?

**Reply by nikriaz** [reply]
I looked for native vue mask. 

And that mask, as I said, is also not ideal. So, maybe you will consider implementing some new features and we all will have a closer to the ideal mask :)
